src/Application.php

```diff
@@ -32,120 +35,117 @@
 
     /**
-     * @var Kernel
-     *
-     * Kernel
+     * @var OutputPrinter
      */
-    private $kernel;
+    private $outputPrinter;
 
     /**
      * Application constructor.
      *
-     * @param string      $rootPath
-     * @param string      $host
-     * @param int         $port
-     * @param string      $environment
-     * @param bool        $debug
-     * @param bool        $silent
-     * @param string      $adapter
-     * @param string      $bootstrapPath
-     * @param string|null $staticFolder
+     * @param LoopInterface       $loop
+     * @param RequestHandler      $requestHandler
+     * @param FilesystemInterface $filesystem
+     * @param ServerContext       $serverContext
+     * @param string              $rootPath
+     * @param string              $bootstrapPath
      *
      * @throws Exception
      */
     public function __construct(
+        LoopInterface $loop,
+        RequestHandler $requestHandler,
+        FilesystemInterface $filesystem,
+        ServerContext $serverContext,
+        OutputPrinter $outputPrinter,
         string $rootPath,
-        string $host,
-        int $port,
-        string $environment,
-        bool $debug,
-        bool $silent,
-        string $adapter,
-        string $bootstrapPath,
-        ?string $staticFolder
+        string $bootstrapPath
     ) {
+        $this->loop = $loop;
+        $this->requestHandler = $requestHandler;
+        $this->filesystem = $filesystem;
+        $this->serverContext = $serverContext;
+        $this->outputPrinter = $outputPrinter;
         $this->rootPath = $rootPath;
-        $this->host = $host;
-        $this->port = $port;
-        $this->environment = $environment;
-        $this->debug = $debug;
-        $this->silent = $silent;
-        $this->adapter = $adapter;
         $this->bootstrapPath = $bootstrapPath;
 
         ErrorHandler::handle();
-        if ($this->debug) {
+        if ($serverContext->isDebug()) {
             umask(0000);
             Debug::enable();
         }
 
-        /*
-         * @var KernelAdapter $adapter
+        /**
+         * @var KernelAdapter
          */
+        $adapter = $serverContext->getAdapter();
         $this->kernel = $adapter::buildKernel(
-            $this->environment,
-            $this->debug
+            $serverContext->getEnvironment(),
+            $serverContext->isDebug()
         );
 
         if (!$this->kernel instanceof AsyncKernel) {
             throw new Exception(sprintf('Your kernel MUST implement %s', AsyncKernel::class));
         }
+    }
 
-        if (!is_null($staticFolder)) {
-            $this->staticFolder = empty($staticFolder)
-                ? $adapter::getStaticFolder($this->kernel)
-                : $staticFolder;
-
-            if (is_string($this->staticFolder)) {
-                $this->staticFolder = '/'.trim($this->staticFolder, '/').'/';
-            }
+    /**
+     * Connect socket.
+     */
+    public function connectSocket()
```

Comment:
Do we really need this public method? Maybe it should be inlined in `run()` as it was before? 🤔